rust/src/buffer.rs

```diff
@@ -159,7 +166,7 @@ impl From<Bytes> for Buffer<u8> {
                     mem::transmute::<*const u8, *const libc::c_void>(bytes.as_ptr()),
                     len * sz,
                 );
-                mem::transmute::<*mut libc::c_void, *const u8>(dst)
+                mem::transmute::<*mut libc::c_void, *mut u8>(dst)
```

Comment:
Again, `transmute` is overkill, a cast would do (check out the [nomicon](https://doc.rust-lang.org/nightly/nomicon/casts.html).